src/test/javascript/CommonJsTest.js

```diff
@@ -78,6 +78,70 @@ QUnit.test('sortDate(x, y)', function(assert) {
+    assert.equal(getPointValue('1%', true), 101, 'Case 1%');
+    assert.equal(getPointValue('-1%', true), 99, 'Case -1%');
+    
+    assert.equal(getPointValue('E -1%', false), 99, 'Case E -1%');
+    assert.equal(getPointValue('E +1%', false), 101, 'Case E +1%');
+    assert.equal(getPointValue('E +100%', false), 200, 'Case E +100%');
+    assert.equal(getPointValue('E -100%', false), 0, 'Case E -100%');
+    
+    assert.equal(getPointValue('E', false), 100, 'Case E');
+    
+    assert.equal(getPointValue('0', false), 100, 'Integer 0');
+    assert.equal(getPointValue('-1', false), 99, 'Integer -1');
+    assert.equal(getPointValue('1', false), 101, 'Integer 1');
+    
+    function isCloseEnough(numberA, numberB) {
+        var tolerance = 0.0001;
+        return Math.abs(numberA - numberB) < tolerance;
+    }
+    assert.ok(isCloseEnough(getPointValue('0.0', false), 100), 'Float 0');
+    assert.ok(isCloseEnough(getPointValue('0.1', false), 100.1), 'Float 0.1');
+    assert.ok(isCloseEnough(getPointValue('-0.1', false), 99.9), 'Float -0.1');
+    assert.ok(isCloseEnough(getPointValue('1.91', false), 101.91), 'Float 1.91');
+    assert.ok(isCloseEnough(getPointValue('-1.22', false), 98.78), 'Float -1.22');
+    assert.ok(isCloseEnough(getPointValue('3.833333', false), 103.833333), 'Float 3.833333');
+    assert.ok(isCloseEnough(getPointValue('-3.833333', false), 96.166667), 'Float -3.833333');
+});
+
+QUnit.test('sortByPoint(a, b)', function(assert) {
+    assert.ok(sortByPoint('N/S', 'N/A') < 0, 'Case N/S less than N/A');
+    assert.ok(sortByPoint('N/S', 'E') > 0, 'N/S more than E');
+    assert.ok(sortByPoint('N/A', 'E +1%') > 0, 'N/A more than E +(-)X%');
+    assert.ok(sortByPoint('N/S', 'E -1%') > 0, 'N/S more than E +(-)X%');
+    
+    assert.ok(sortByPoint('0%', '0%') === 0, 'Case 0% equal 0%');
+    assert.ok(sortByPoint('-1%', '0%') > 0, 'Case 0% less than every X%');
+    assert.ok(sortByPoint('1%', '0%') > 0, 'Case 0% less than every X%');
+    assert.ok(sortByPoint('3%', '-2%') > 0, 'Case 3% more than -2%');
+    
+    assert.ok(sortByPoint('E +1%', 'E') > 0, 'Case E +1% more than E');
+    assert.ok(sortByPoint('E -1%', 'E') < 0, 'Case E -1% less than E');
+    assert.ok(sortByPoint('E +33%', 'E -23%') > 0, 'Case E +33% more than E -23%');
+    
+    assert.ok(sortByPoint('0', '-1') > 0, 'Case Integer 0 more than -1');
+    assert.ok(sortByPoint('1', '0') > 0, 'Case Integer 1 more than 0');
+    assert.ok(sortByPoint('2', '-3') > 0, 'Case Integer 2 more than -3');
+    
+    assert.ok(sortByPoint('0.0', '1.0') < 0, 'Case Float 0.0 less than 1.0');
+    assert.ok(sortByPoint('0.3', '-1.1') > 0, 'Case Float 0.3 more than -1.1');
+    assert.ok(sortByPoint('0.3', '0.33338') < 0, 'Case Float 0.3 less than 0.33338');
+    assert.ok(sortByPoint('-4.33333', '-4.5') > 0, 'Case Float -4.33333 more than -4.5');
+    
+    assert.ok(sortByPoint('NotNumber', 'Random') === 0, 'Equality for NaN');
+});
+
```

Comment:
@damithc Yes. There are browser test cases for the sorting functionality, but not so comprehensive like this. I shall update it also.